File: Frontend/skillforge/src/lib/types/HomePageData.ts
import type TagLink from "./TagLinkType";
import type TopArticleItemType from "./TopArticleItemType";
import type UserLink from "./UserLinkType";

export default interface HomePageData {
    TopArticles: TopArticleItemType[],
    TopUsers: UserLink[],
    TopTags: TagLink[],
}

File: Backend/SkillForge/SkillForge/Areas/Admin/Models/DTOs/HomePageData.cs
using SkillForge.Areas.Admin.Models.DTOs.Article;

namespace SkillForge.Areas.Admin.Models.DTOs;

public class HomePageData
{
    public List<ArticleCard> LatestArticles { get; set; }

    public List<TopArticleItem> TopArticles { get; set; }
    
    public List<UserLink> TopUsers { get; set; }
    
    public List<TagLink> TopTags { get; set; }
}
